ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-blocks/embeddedActivityBlock/embeddedActivityBlock.component.ts

```diff
@@ -1,63 +1,91 @@
+    selector: 'ddp-embedded-activity-block',
+    templateUrl: './embeddedActivityBlock.component.html',
+    styleUrls: ['./embeddedActivityBlock.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class EmbeddedActivityBlockComponent implements OnInit {
-  @Input() instance: ActivityInstance;
-  @Input() readonly: boolean;
-  @Input() validationRequested: boolean;
-  @Input() studyGuid: string;
-  @Output() componentBusy = new EventEmitter<boolean>(true);
-
-  public activity: ActivityForm;
-  private readonly LOG_SOURCE = 'EmbeddedActivityBlockComponent';
-
-  constructor(private activityServiceAgent: ActivityServiceAgent,
-              private changeDetector: ChangeDetectorRef,
-              private submitAnnouncementService: SubmitAnnouncementService,
-              private logger: LoggingService) { }
-
-  ngOnInit(): void {
-    this.getActivity();
-    this.submitAnnouncementService.submitAnnounced$.pipe(
-        tap(() => this.componentBusy.emit(true)),
-        concatMap(() => this.activityServiceAgent.flushForm(this.studyGuid, this.instance.instanceGuid)),
-        catchError(err => {
-            this.logger.logError(this.LOG_SOURCE, 'An error during completing an activity', err);
-            return EMPTY;
-        }),
-        take(1),
-        finalize(() => this.componentBusy.emit(false))
-    ).subscribe();
-  }
-
-  public getSectionId(index: number, {name}: ActivitySection): string {
-    return `${index}_${name}`;
-  }
-
-  private getActivity(): void {
-    this.activityServiceAgent.getActivity(of(this.studyGuid), of(this.instance.instanceGuid))
-      .pipe(
-        catchError(err => {
-          this.logger.logError(this.LOG_SOURCE, 'An error during getting a full activity', err);
-          return EMPTY;
-        }),
-        take(1))
-      .subscribe((activity: ActivityForm) => {
-        this.activity = activity;
-        this.changeDetector.detectChanges();
-      });
-  }
+    @Input() instance: ActivityInstance;
+    @Input() readonly: boolean;
+    @Input() validationRequested: boolean;
+    @Input() studyGuid: string;
+    @Output() componentBusy = new EventEmitter<boolean>(true);
+    @Output() deletedActivity = new EventEmitter<string>();
+
+    public activity: ActivityForm;
+    private readonly LOG_SOURCE = 'EmbeddedActivityBlockComponent';
+
+    constructor(private activityServiceAgent: ActivityServiceAgent,
+                private changeDetector: ChangeDetectorRef,
+                private submitAnnouncementService: SubmitAnnouncementService,
+                private logger: LoggingService) {
+    }
+
+    ngOnInit(): void {
+        this.getActivity();
+        this.flushFormOnSubmit();
+    }
+
+    public getSectionId(index: number, {name}: ActivitySection): string {
+        return `${index}_${name}`;
+    }
+
+    public deleteInstance(): void {
+        this.activityServiceAgent.deleteActivityInstance(this.studyGuid, this.instance.instanceGuid).pipe(
```

Comment:
just wondering if we need to ask the user to confirm s/he wants to delete (protect from occasional clicks). It makes sense  sometimes if the user already filled the fields with some information and doesn't want to lose it

Reply:
Good point.
When we discussed it with Marco and Yufeng, we agreed that it should be similar like it is implemented in RGP study.
There is no any deleting confirmation there for that case.

Reply:
This is a good question. I don't see how we would not want confirmation (OK. Maybe if the child activity is totally empty/never edited, don't see any harm in deleting without confirmation). But otherwise, the data is precious and we should make sure the user is sure. 
Modal activity deletion I think is a good model.

Reply:
Applied a common delete dialog for both modal/embedded activity blocks (at 8defb272911444c7b77c9affc361e16198b77a50)